f/flowable-engine-all-modules-generic/flowable-engine_module_ubuntu_18.04.sh

```diff
@@ -0,0 +1,125 @@
+# or any list of modules under package  https://github.com/flowable/flowable-engine
+# Script License   : Apache License, Version 2 or later
+# Maintainer       : Arumugam N S <asellappen@yahoo.com> / Priya Seth<sethp@us.ibm.com>
+#
+# Disclaimer       : This script has been tested in non-root mode on given
+# ==========  platform using the mentioned version of the package.
+#             It may not work as expected with newer versions of the
+#             package and/or distribution. In such case, please
+#             contact "Maintainer" of this script.
+#
+# ----------------------------------------------------------------------------
+#!/bin/bash
+
+
+export REPO=https://github.com/flowable/flowable-engine
+
+#Default tag flowable-6.6.0
+if [ -z "$1" ]; then
+  export VERSION="flowable-6.6.0"
+else
+  export VERSION="$1"
+fi
+
+#Default testing on jdk8
+if [ -z "$2" ]; then
+  export JDK="openjdk-8-jdk"
+else
+  export JDK="$2"
+fi
+
+#Module options need to be passed,otherwise Raise exception
+if [ -z "$3" ]; then
+ echo "Enter the module name to be build or List of modules to be build\n
+ eg :-
+      flowable-engine_module_ubuntu_18.04.sh '' '' 'flowable-ui-app'
+      or
+      flowable-engine_module_ubuntu_18.04.sh '' '' 'flowable-batch-service flowable-batch-service-api flowable-bpmn-converter .....' "
+  exit
+else
+  export MODULElIST="$3"
+fi
+
+#Default installation
+sudo apt-get update
+sudo apt-get install -y apt-utils
+sudo apt-get install wget git -y
+
+#Fro rerunning build
+if [ -d "flowable-engine" ] ; then
+  rm -rf flowable-engine
+fi
+
+# run tests with java 11 or jdk 8
+sudo apt-get install -y ${JDK}
+jret=$?
+if [ $jret -eq 0 ] ; then
+  echo "Sucessfully installed JDK  ${JDK} "
+else
+  echo "Failed to install JDK  ${JDK} "
+  exit
+fi
+
+#Setting JAVA_HOME
+export folder=`echo ${JDK}  | grep -oP '(?<=openjdk-).*(?=-jdk)'`
+export JAVA_HOME=/usr/lib/jvm/java-${folder}-openjdk-ppc64el/
+
+
+sudo apt install  -y maven
+
+mvn -v
+
+
+git clone ${REPO}
+cd flowable-engine
+git checkout ${VERSION}
+ret=$?
+if [ $ret -eq 0 ] ; then
+  echo  "${VERSION} found to checkout"
+else
+  echo  "${VERSION} not found"
+  exit
+fi
+#goto module path
+cd modules
+for MODULE in ${MODULElIST};
+do
+ # Check module path
+    cd ${MODULE}
+    ret=$?
+    if [ $ret -eq 0 ] ; then
+      echo  "${MODULE} module found in the package to build and started building ..."
+    else
+      echo  "${MODULE} module not found in the package ..."
```

Comment:
Don't we need to cd to parent directory here ?

Reply:
Not required ,Since the build and test are done in the module level then pom.xml from module is being used .(ex. /flowable-engine/modules/flowable-task-service-api/pom.xml). 
Each module having pom.xml and it is being used for building and testing respective module.

Reply:
pl. refer test case 11,This is being tested .

Reply:
ok